src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -251,7 +251,54 @@ internal bool IsDNSCachingBeforeRedirectSupported
         // If the context is expiring within the next 10 mins, then create a new context, irrespective of if another thread is trying to do the same.
         private static readonly TimeSpan _dbAuthenticationContextUnLockedRefreshTimeSpan = new TimeSpan(hours: 0, minutes: 10, seconds: 00);
 
-        private static HashSet<int> s_transientErrors = new HashSet<int>();
+        // The errors in the transient error set are contained in
+        // https://azure.microsoft.com/en-us/documentation/articles/sql-database-develop-error-messages/#transient-faults-connection-loss-and-other-temporary-errors
+        private static readonly HashSet<int> s_transientErrors = new HashSet<int>
```

Comment:
Dumb question - why are we using a hashset here instead of just a statically defined array?

Reply:
Likely to indicate that the error codes are meant to be unique. It might also be to align with the SqlConfigurableRetryFactory class' definition of transient error numbers.

Reply:
I guess I'd look at a static array of error codes and just assume that they are unique. I could see using the hashset if we allowed the user to add more errors to the list.
Again, outside the scope of this PR, so I won't make a fuss about it.